andela/jubilee-backend
# #167733644 signup/Registration Validation Errors

### What does this PR do?
Validate user inputs and throws errors whenever user does not meet input requirements.

### Description of Task to be completed?
- Install needed validation depenencies (e.g. @hapi/joi package)
- Create user validation middleware
-  Add validation class for user input at signup
- Update routes
- Write the test for validations
- Test validations (on success and on failure)

### How should this be manually tested?
- Clone repository
- update or copy .env file from .env.sample
- Run `npm run prestart` to create tables in your database.
- Run  npm run start:dev on local root directory
- Test router on Postman with POST: 'localhost:3000/api/auth/signup'
- Run a test case JSON script on Postman body, eg:
```
{
	"firstName": "Tony",
	"lastName": "Marshal",
	"email": "tony@marshal.com",
	"password": "tmonarqA1.",
	"gender": "male",
	"street": "Abage",
	"city": "Lekki",
	"state": "Lagos",
	"country": "Nigeria",
	"birthdate": "2009-08-20",
	"phoneNumber": "(+234) 098-3442-1889",
	"companyName": "Jubilee"
}
```
- Omit some fields to check that the validation script works on inputs not given by the user.

### Any background context you want to provide?
None

### What are the relevant pivotal tracker stories?
[#167733644](https://www.pivotaltracker.com/story/show/167733644)
